### PR TITLE
Add input-wrap form directive

### PR DESCRIPTION
**Pull request by calebthompson**
This should DRY up a lot of the form building business.

### CODE REVIEW

**Comment by chriskacerguis** [comment]
lgtm

**Comment by calebthompson** [comment]
I'm gonna wait on @sonejah21 and @AllenBW to see if they think this is a positive refactor.

File: public/templates/partials/common/question_input.html [hunk 1]
```diff
@@ -1,46 +1,22 @@
-<div ng-switch on="qiCtrl.question.field_type">
-
-  <div ng-switch-when="text" class="form-group" ng-class="{'has-error': qiCtrl.hasError()}">
-    <label class="col-sm-2 control-label" for="{{qiCtrl.fieldId()}}">{{qiCtrl.question.label}}</label>
-
-    <div class="col-sm-6 control-input">
+<form-wrap for="{{qiCtrl.fieldId()}}" label={{qiCtrl.question.label}} errorPresent="qiCtrl.hasError()">
```
Comment (stackus) [inline public/templates/partials/common/question_input.html]: I think you meant input-wrap here.

File: public/templates/partials/forms/input_wrap.html [hunk 2]
```diff
@@ -0,0 +1,4 @@
+<div class="form-group" ng-class="{'has-error': errorPresent}">
+  <label class="col-sm-3 control-label" for="{{for}}">{{label}}</label>
```
Comment (stackus) [inline public/templates/partials/forms/input_wrap.html]: These two values should be once-bound. To do this add `::` to the start of the variable

example: `{{::for}}`

Comment (calebthompson) [inline public/templates/partials/forms/input_wrap.html]: Just in this template, or everywhere the directive is used?

File: public/templates/partials/common/question_input.html (hunk 1 above)
Comment (calebthompson) [inline public/templates/partials/common/question_input.html]: Sure did, thanks!

File: public/templates/partials/forms/input_wrap.html (hunk 2 above)
Comment (stackus) [inline public/templates/partials/forms/input_wrap.html]: Just here. Doing this should reduce the number of additional watches by two every time this directive is used.

**Comment by calebthompson** [comment]
@stackus thanks. I addressed both of those points.

**Comment by masonforest** [comment]
:+1:

**Pull request merged**